Add changelog entry for 9.4.0 (#3966)
citusdata/citus: CHANGELOG.md

> ### citus v9.4.0 (July 1, 2020) ###

Author: ```suggestion
### citus v9.4.0 (July 28, 2020) ###
```

* Improves COPY by honoring max_adaptive_executor_pool_size config

* Adds support for insert into local table select from distributed table

* Adds support to partially push down tdigest aggregates

* Enables joins between local tables and CTEs

* Adds showing query text in EXPLAIN output when explain verbose is true

* Adds support for showing CTE statistics in EXPLAIN ANALYZE

* Adds support for showing amount of data received in EXPLAIN ANALYZE

* Introduces downgrade paths in migration scripts

* Avoids returning incorrect results when changing roles in a transaction

* Fixes `ALTER TABLE IF EXISTS SET SCHEMA` with non-existing table bug

* Fixes `CREATE INDEX CONCURRENTLY` with no index name on a postgres table bug

* Fixes a bug that could cause crashes with certain compile flags

* Fixes a bug with lists of configuration values in ALTER ROLE SET statements

* Fixes a bug that occurs when coordinator is added as a worker node

* Fixes a crash because of overflow in partition id with certain compile flags

> * Fixes a crash that may happen if no worker nodes added

Reviewer: ```suggestion
* Fixes a crash that may happen if no worker nodes are added
```

* Fixes a crash that occurs when inserting implicitly coerced constants

* Fixes a crash when aggregating empty tables

* Fixes a memory leak in subtransaction memory handling

* Fixes crash when using rollback to savepoint after cancellation of DML

* Fixes deparsing for queries with anonymous column references

* Fixes distribution of composite types failing to include typemods

* Fixes explain analyze on adaptive executor repartitions

* Fixes possible error throwing in abort handle

* Fixes segfault when evaluating func calls with default params on coordinator

* Fixes several EXPLAIN ANALYZE issues

* Fixes write queries with const expressions and COLLATE in various places

> * Fixes wrong cancellation message about distributed deadlock

Reviewer: ```suggestion
* Fixes wrong cancellation message about distributed deadlocks
```

### citus v9.4.0 (July 28, 2020) ###

* Improves COPY by honoring max_adaptive_executor_pool_size config

* Adds support for insert into local table select from distributed table

* Adds support to partially push down tdigest aggregates

> * Supports binary encoded rslts from workers using citus.enable_binary_protocol

Author: ```suggestion
* Adds support for receiving binary encoded results from workers using
   citus.enable_binary_protocol
```

* Adds support to partially push down tdigest aggregates

* Supports binary encoded rslts from workers using citus.enable_binary_protocol

> 

Author: I would like to write 
`Adds support for receiving binary encoded results from workers using citus.enable_binary_protocol` but it is too long, what about the above alternative @hanefi & @JelteF

Reviewer 2: Just put it on two lines

Reviewer 2: Don't abreviate results as rslts at least.

Author: afaik, we don't prefer multi line changelog entries, what do you think ?

Reviewer 2: I thought in the last discussion the consensus was that our changelog was not very useful to users, because the items were too short. I think multiline entries are a good way to fix that, while keeping the 80 character line limit imposed by debian packaging.

Author: it totally makes sense, but I'm still not much sure as it is a decision more than this pr :/

Reviewer: Some of our automated scripts can potentially break as they do not support multiline entries yet. However we can manually fix it this time.

This can also be a nice practice to figure out what needs to be updated on those scripts.

I'm ok with having a multiline entry here.

Author: addressed
